clients/src/test/java/org/apache/kafka/common/requests/OffsetFetchRequestTest.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.kafka.common.requests.OffsetFetchResponse.PartitionData;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.kafka.common.requests.AbstractResponse.DEFAULT_THROTTLE_TIME;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+public class OffsetFetchRequestTest {
+
+    private final String topicOne = "topic1";
+    private final int partitionOne = 1;
+    private final String topicTwo = "topic2";
+    private final int partitionTwo = 2;
+    private final String groupId = "groupId";
+
+    private OffsetFetchRequest.Builder builder;
+    private List<TopicPartition> partitions;
+
+    @Before
+    public void setUp() {
+        partitions = Arrays.asList(new TopicPartition(topicOne, partitionOne),
+                                   new TopicPartition(topicTwo, partitionTwo));
+        builder = new OffsetFetchRequest.Builder(
+            groupId,
+            partitions
+        );
+    }
+
+    @Test
+    public void testConstructor() {
+        assertFalse(builder.isAllTopicPartitions());
+        Map<Errors, Integer> expectedErrorCounts = new HashMap<>();
+        int throttleTimeMs = 10;
+        expectedErrorCounts.put(Errors.NONE, 2);
+
+        Map<TopicPartition, PartitionData> expectedData = new HashMap<>();
+        for (TopicPartition partition : partitions) {
+            expectedData.put(partition, new PartitionData(
+                OffsetFetchResponse.INVALID_OFFSET,
+                Optional.empty(),
+                OffsetFetchResponse.NO_METADATA,
+                Errors.NONE
+            ));
+        }
+
+        for (short version = 0; version < OffsetFetchRequestData.SCHEMAS.length; version++) {
+            OffsetFetchRequest request = builder.build(version);
+            assertFalse(request.isAllPartitions());
+            assertEquals(groupId, request.groupId());
+            assertEquals(partitions, request.partitions());
+
+            OffsetFetchResponse response = request.getErrorResponse(throttleTimeMs, Errors.NONE);
+            assertEquals(Errors.NONE, response.error());
+            assertFalse(response.hasError());
+            assertEquals(Collections.singletonMap(Errors.NONE, 1), response.errorCounts());
+
+            if (version < 2) {
+                assertEquals(expectedData, response.responseData());
+            }
+
+            if (version > 2) {
+                assertEquals(throttleTimeMs, response.throttleTimeMs());
+            } else {
+                assertEquals(DEFAULT_THROTTLE_TIME, response.throttleTimeMs());
+            }
+        }
+    }
+
+    @Test(expected = UnsupportedVersionException.class)
+    public void testConstructorFailForUnsupportedAllPartition() {
```

Comment:
nit: We're not really testing the constructor here. A better name might be something like `testAllPartitionsNotSupportedForOlderVersions`. If we modify this code to use `assertThrows`, then we can test both versions 0 and 1.